R/compat-types-check.R

```diff
@@ -85,12 +86,19 @@ check_string <- function(x,
 
 check_number <- function(x,
                          ...,
+                         allow_decimal = FALSE,
```

Comment:
Should we make this true by default? I think most people would say that 10.5 is a number.

Reply:
The thing is that, at least in the sort of packages we work on, it's much more common to take integer inputs than real numbers (e.g. when we take a length or size, a number of rows, an iteration number, etc). So I think we need a function with the right default for these cases. We could export another function for this but the name would be longer: `check_round_number()`.

Reply:
`check_number_decimal()` + `check_number_whole()` might work? Then autocomplete would remind you of the options, and the longer names aren't that important since the call will usually be short.

Reply:
I've split the functions in the two variants that you suggested. `_decimal` allows infinite values by default, and `_whole` hard-codes `allow_infinite` to `FALSE`.